src/DotNet.Status.Web/Controllers/GitHubHookController.cs

```diff
@@ -53,22 +59,187 @@ private async Task EnsureLabelsAsync()
             );
         }
 
+        private readonly SimpleJsonSerializer _serializer = new SimpleJsonSerializer();
+
+        private async Task<T> DeserializeGitHubWebHook<T>()
+        {
+            // Octokit has a serializer and types built for github rest apis, use them
+            string data;
+            using (var reader = new StreamReader(Request.Body))
+            {
+                data = await reader.ReadToEndAsync();
+            }
+
+            var payload = _serializer.Deserialize<T>(data);
+            return payload;
+        }
+
+        [GitHubWebHook(EventName = "pull_request_review_comment")]
+        public async Task<IActionResult> PullRequestReviewComment()
+        {
+            var payload = await DeserializeGitHubWebHook<PullRequestCommentPayloadWithChanges>();
+
+            string repo = payload.Repository.Owner.Login + "/" + payload.Repository.Name;
+            int number = payload.PullRequest.Number;
+            _logger.LogInformation("Received webhook for pull request {repo}#{number}", repo, number);
+
+            string title = payload.PullRequest.Title;
+            string uri = payload.Comment.HtmlUrl;
+            string username = payload.PullRequest.User.Login;
+            DateTimeOffset date = payload.PullRequest.UpdatedAt;
+            try
+            {
+                bool sent = false;
+                switch (payload.Action)
+                {
+                    case "created":
+                        sent = await _issueMentionForwarder.HandleIssueBody(null, payload.Comment.Body, title, uri, username, date);
+                        break;
+                    case "edited" when !string.IsNullOrEmpty(payload.Changes.Body?.From):
+                        sent = await _issueMentionForwarder.HandleIssueBody(payload.Changes.Body.From, payload.Comment.Body, title, uri, username, date);
+                        break;
+                }
+
+                if (sent)
+                {
+                    _logger.LogInformation("Sent teams notification for pull request {repo}#{number}", repo, number);
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogWarning(ex, "Unable to send teams notification for pull request {repo}#{number}", repo, number);
+            }
+
+            return Ok();
+        }
+
+        [GitHubWebHook(EventName = "issue_comment")]
+        public async Task<IActionResult> IssueComment()
```

Comment:
This is a fairly large chunk of duplicatd code... is there nothing more we can share?

Reply:
I didn't like it a ton. I think I shared as much as I could. This duplicated code is just extracting the relevant values from the payload and checking the "action"

Reply:
Bleh, that's true. looks good then. :-)